grommet/grommet
# Allow onClickOutside for Layer when modal={false}

#### What does this PR do?
Allows `onClickOutside` prop to work on Layer when `modal={false}`. Previously, this only functioned when modal={true}. Referenced the behavior that `DropContainer` was applying as it already provides the desired behavior.

#### Where should the reviewer start?
src/js/components/Layer/LayerContainer.js

#### What testing has been done on this PR?
Updated jest test and tested on storybook.

#### How should this be manually tested?
On Layer Target story example, switch modal to "false" using the toggle. Then, open the Layer and click outside of it. Layer should close.

#### Any background context you want to provide?
Design use cases for a non-modal Layer where `onClickOutside` still functions created a need for this functionality. For example, a Sidebar layer that opens over screen content that does not prohibit the user from interacting with the rest of the page should still be able to be closed by clicking outside.

#### What are the relevant issues?
Related to https://github.com/grommet/grommet/issues/3771

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
`onClickOutside` is now supported for Layer with modal={false}.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.
